### PR TITLE
Week 2 task, create comments feature

### PR DESCRIPTION
**Pull request by leonyeo**
Completed step 1, edit servlet

### CODE REVIEW

**Comment by leonyeo** [comment]
Completed step 2, fetch data from servlet

**Comment by leonyeo** [comment]
Completed step 3, fetch comments as JSON and display

**Comment by leonyeo** [comment]
Completed step 4, create comments feature (form to post comments, fetch and display comments)

**Comment by leonyeo** [comment]
Save comment to datastore instead of local arraylist

**Comment by leonyeo** [comment]
Load comments from datastore

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -14,19 +14,47 @@
...
-@WebServlet("/data")
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    DatastoreService comments;
```
Comment (henryken) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The variable name doesn't represent the actual type.
I would suggest "datastore" or "datastoreService".

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -14,19 +14,47 @@
...
+import java.util.Date;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comment")
```
Comment (henryken) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:36]: Best practice of REST API resource name should be plural.
Further reference can be found here:
https://google.aip.dev/122#collection-identifiers

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -14,19 +14,47 @@
...
+    @Override
+    public void init() {
+        this.comments = DatastoreServiceFactory.getDatastoreService();
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("application/json;");
+        Gson gson = new Gson();
+        response.getWriter().println(gson.toJson(this.comments));
```
Comment (henryken) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: `this.comments` doesn't retrieve anything?
You might miss the [Loading Entities](https://github.com/leonyeo/google-sps/blob/master/walkthroughs/week-2-server/step-5-datastore-walkthrough.md#loading-entities) section?

File: portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.Date;
+
+// Class containing comment data
+public final class Comment {
```
Comment (henryken) [inline portfolio/src/main/java/com/google/sps/data/Comment.java:20]: Where do you use this class?

**Comment by henryken** [comment]
Next time, it would be good if you notify if you're still pushing changes to the PR so that I don't review outdated code.
Please let me know when you're done with the changes.

**Pull request merged**